### PR TITLE
Extend and validate ApplicationConfiguration proto

### PR DESCRIPTION
**Pull request by daviddrysdale**

### CODE REVIEW

File: oak/common/app_config.cc [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+#include "asylo/util/logging.h"
+
+namespace oak {
+
+bool ValidApplicationConfig(ApplicationConfiguration* config) {
+  if (config == nullptr) {
+    return false;
+  }
+  // Check name uniqueness and count pseudo-nodes.
+  using fqpn = std::pair<std::string, std::string>;
```
Comment (tiziano88) [inline oak/common/app_config.cc]: Could you clarify what the components mean?

File: oak/common/app_config.cc [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+#include "oak/common/app_config.h"
+
+#include <set>
+#include <utility>
+
+#include "asylo/util/logging.h"
+
+namespace oak {
+
+bool ValidApplicationConfig(ApplicationConfiguration* config) {
```
Comment (tiziano88) [inline oak/common/app_config.cc]: Could you pass config by const reference instead?

File: oak/common/app_config.cc [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  if (config == nullptr) {
+    return false;
+  }
+  // Check name uniqueness and count pseudo-nodes.
+  using fqpn = std::pair<std::string, std::string>;
+  std::set<fqpn> in_ports;
```
Comment (tiziano88) [inline oak/common/app_config.cc]: Consider https://en.cppreference.com/w/cpp/container/multimap instead?

File: oak/common/app_config.cc (hunk 2 above)
Comment (daviddrysdale) [reply oak/common/app_config.cc]: Good point, done.

File: oak/common/app_config.cc (hunk 1 above)
Comment (daviddrysdale) [reply oak/common/app_config.cc]: Done.

File: oak/common/app_config.cc (hunk 3 above)
Comment (daviddrysdale) [reply oak/common/app_config.cc]: Not sure how that would help &ndash;  the <node,port> name should be globally unique, and we need to track direction for the checks later.

Comment (tiziano88) [reply oak/common/app_config.cc]: Ah got it, you need to be able to look up by the entire pair, not just the first component.

**Pull request merged**